doc/designs/fido2.md

```diff
@@ -0,0 +1,239 @@
+`Users can manage their own FIDO2 key mappings`
+
+#### Privilege
+
+- New privilege `FIDO2 Administrators` with the permissions `System: Modify FIDO2 Configuration` and `System: Manage FIDO2 Key Mappings`.
+
+By default only members of the admins group are allowed to modify the FIDO2
+settings or another user's FIDO2 keys.
+
+## Implementation
+
+### LDAP schema
+
+New objectclass and attribute for the FIDO2 configuration object:
+```
+attributeTypes: ( 2.16.840.1.113730.3.8.24.26 NAME 'ipaRequireUserVerif' DESC 'require FIDO2 user verification' SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 SINGLE-VALUE X-ORIGIN 'IPA v4.10')
+objectclasses: ( 2.16.840.1.113730.3.8.24.8 NAME 'ipaFido2ConfigObject' DESC 'IPA FIDO2 global config options' AUXILIARY MUST ipaRequireUserVerif X-ORIGIN 'IPA v4.10')
+```
+
+New objectclass and attribute for the FIDO2 key mapping:
+```
+attributeTypes: ( 2.16.840.1.113730.3.8.24.27 NAME 'ipafido2key' DESC 'FIDO2 key mapping' EQUALITY caseExactMatch SYNTAX 1.3.6.1.4.1.1466.115.121.1.15 X-ORIGIN 'IPA v4.10' )
+objectclasses: ( 2.16.840.1.113730.3.8.24.9 NAME 'ipaFido2User' DESC 'IPA FIDO2 user' AUXILIARY MAY ipafido2key X-ORIGIN 'IPA v4.10')
+```
+
+### Indices
+
+No need to add a new index for ipafido2keys as the search performed by SSSD
+will use a filter based on the user uid.
+
+
+## Feature Management
+
+### UI
+
+- A new tab will be added below "Policy", at the same level as `Host-Based Access Control`, `Sudo`, `SELInux User Maps`, `Password Policies` and `Kerberos Ticket Policy`, with the label `FIDO2 Configuration`.
+
+It will allow to configure the attribute `Require User Verification`, with a radio button: `on`, `off` or `default`.
+
+- In the `User` facet, a new field will be added, below `SSH public keys`, with the label `FIDO2 key mapping`, and will display the values, or allow to add a new value.
+
+Note: since the Web browser my be running on a non-enrolled host without
+the required packages, the WebUI will probably need specific javascript code
+to register a key by inserting it on the machine where the browser is
+running.
+
+Investigations TBD regarding the possible solutions.
+
+
+### CLI
+
+| Command |	Options | Description |
+| --- | ----- | --- |
+| | **FIDO2 configuration** |
+| fido2config-show | | This command displays the FIDO2 settings |
+| fido2config-mod | --require-user-verif=['on', 'off', 'default'] | This command modifies the FIDO2 settings |
+| | **User Mapping** | |
+| user-add-fido2key | LOGIN [FIDO2KEY...] | This command does not require the FIDO2 key to be inserted and can directly add the mapping data, obtained through another mean (for instance through fido2_child) |
+| user-add-fido2key | LOGIN --register [--cose-type=['es256', 'rs256', 'eddsa']] [--require-user-verif] | This command requires the insertion of the FIDO2 key, performs the registration with the specified cose type + user verification requirement, and adds the mapping data to the user entry |
```

Comment:
Should this at least be --require-user-verify if not --require-user-verification? It seems cut off w/o the 'y'. This may be a FIDO-world convention, I don't know.

Reply:
Replaced with require-user-verification